src/__experimental__/components/date/date-picker.spec.js

```diff
@@ -0,0 +1,139 @@
+import moment from 'moment';
+import MockDate from 'mockdate';
+import React from 'react';
+import TestRenderer from 'react-test-renderer';
+import { mount, shallow } from 'enzyme';
+import DayPicker from 'react-day-picker';
+import Portal from '../../../components/portal/portal';
+import DatePicker from './date-picker.component';
+import StyledDayPicker from './day-picker.style';
+import classicTheme from '../../../style/themes/classic';
+
+const inputElement = { value: '12-12-2012', getBoundingClientRect: () => ({ left: 0, bottom: 0 }) };
+const firstDate = '2019-02-02';
+const secondDate = '2019-02-08';
+const currentDate = moment().toDate();
+
+describe('DatePicker', () => {
+  let wrapper;
+
+  describe('when rendered with an "inputElement" prop', () => {
+    beforeEach(() => {
+      wrapper = render({ selectedDate: currentDate, inputElement }, mount);
+    });
+
+    it('should render a "Portal" component', () => {
+      expect(wrapper.find(Portal)).toHaveLength(1);
+    });
+
+    it('should render a "DayPicker" component inside a "Portal"', () => {
+      expect(wrapper.find(Portal).contains(DayPicker)).toBe(true);
+    });
+  });
+
+  describe('when rendered with "minDate" prop', () => {
+    beforeEach(() => {
+      wrapper = render({ inputElement, minDate: firstDate }, mount);
+    });
+
+    it(`should pass to the "DayPicker" component the "disabledDays" 
+        prop containing an object with "before" property`, () => {
+      const disabledDays = [{ before: moment(firstDate).toDate() }];
+      expect(wrapper.find(DayPicker).props().disabledDays).toEqual(disabledDays);
+    });
+  });
+
+  describe('when rendered with "maxDate" prop', () => {
+    beforeEach(() => {
+      wrapper = render({ inputElement, maxDate: secondDate }, mount);
+    });
+
+    it(`should pass to the "DayPicker" component the "disabledDays" 
+        prop containing an object with "after" property`, () => {
+      const disabledDays = [{ after: moment(secondDate).toDate() }];
+      expect(wrapper.find(DayPicker).props().disabledDays).toEqual(disabledDays);
+    });
+  });
+
+  describe('when rendered with both "minDate" and "maxDate" props', () => {
+    beforeEach(() => {
+      wrapper = render({ inputElement, minDate: firstDate, maxDate: secondDate }, mount);
+    });
+
+    it(`should pass to the "DayPicker" component the "disabledDays" 
+        prop containing an object with both "before" and "after" properties`, () => {
+      const disabledDays = [{ before: moment(firstDate).toDate() }, { after: moment(secondDate).toDate() }];
+      expect(wrapper.find(DayPicker).props().disabledDays).toEqual(disabledDays);
+    });
+  });
+
+  describe('when the "onDayClick" prop have been triggered', () => {
+    let handleDateSelectFn;
+
+    beforeEach(() => {
+      handleDateSelectFn = jest.fn();
```

Comment:
> Define with `const` here rather than `let` outside the `beforeEach`?
> 
> Doesn't look like you're redefining this anywhere?

if we'll define it with a `const`, then we'll share the same function between tests, which could lead to false positives

Reply:
> Also, minor thing, but could rename to `handleDateSelect` for simpler syntax in your `render` call on line 75.

I like to emphasize that it's a jest mocked function